### PR TITLE
doc/dev: update log_based_pg.rst, fix some display problem

### PR DESCRIPTION
**Pull request by Liuchang0812**
Signed-off-by: liuchang0812 <liuchang0812@gmail.com>

### CODE REVIEW

File: doc/dev/osd_internals/log_based_pg.rst [hunk 1]
```diff
@@ -81,38 +81,27 @@ cannot.  It would be really annoying if we needed to have two entire
 implementations of PrimaryLogPG, one for each of the two, if there are
 really only a few fundamental differences:
 
-  1. How reads work -- async only, requires remote reads for ec
-  2. How writes work -- either restricted to append, or must
-     write aside and do a tpc
-  3. Whether we choose the oldest or newest possible head entry
-     during peering
-  4. A bit of extra information in the log entry to enable rollback
+1. How reads work -- async only, requires remote reads for ec
```
Comment (tchaikov) [inline doc/dev/osd_internals/log_based_pg.rst]: could use
```rst
#. item 1
#. item 2
```

for ordered list

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: OK~ i'm working on this

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: done. I edit this on github's online editor. Need I squash these commits?

By the way. The ceph docuemnt website is broken. I sent a email to dev-maillist yesterday. It's still broken now.

File: doc/dev/osd_internals/log_based_pg.rst [hunk 2]
```diff
@@ -81,40 +81,28 @@ cannot.  It would be really annoying if we needed to have two entire
...
-  3. Various bits of the write pipeline disallow some operations based
-     on pool type -- like omap operations, class operation reads, and
-		 writes which are not aligned appends (officially, so far)
-     for ec
-  4. Misc other kludges here and there
-
-PGBackend and PGTransaction enables abstraction of differences 1, 2,
+#. All of the stats and metadata for objects
+#. The high level locking rules for mixing client IO with recovery and scrub
+#. The high level locking rules for mixing reads and writes without exposing uncommitted state (which might be rolled back or forgotten later)
```
Comment (tchaikov) [inline doc/dev/osd_internals/log_based_pg.rst]: could you wrap the line at 80 chars?

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: how to make it? seems that RST‘s list doesn't support this.

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: i find a method http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#line-blocks

i will fix this tomorrow

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: i find a method http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#line-blocks

i will fix this tomorrow

Comment (tchaikov) [inline doc/dev/osd_internals/log_based_pg.rst]: @Liuchang0812 for instance, you can simply put
```
#. How writes work -- either restricted to append, or must write aside
   and do a tpc
```
sphinx will render it just right. i tested locally. you can try using `admin/build-doc` to verify the generated html.

Comment (Liuchang0812) [inline doc/dev/osd_internals/log_based_pg.rst]: yes  it works.

but Bullet Lists can't break lines, I use the line-blocks only for "-". Take a look please

File: doc/dev/osd_internals/log_based_pg.rst (hunk 1 above)
Comment (tchaikov) [inline doc/dev/osd_internals/log_based_pg.rst]: happy new year to you!

**Comment by Liuchang0812** [comment]
thanks for reviewing

**Pull request merged**